src/Modal/Modal.js

```diff
@@ -7,43 +7,40 @@ import { Link } from 'react-router-dom';
 import './Modal.scss';
 
 class Modal extends Component {
-  constructor() {
-    super();
+  constructor({throwError}) {
```

Comment:
Holding the state and the functions at the top, and passing them up and down is ... a brain stretch :) but we got there!